torch/_functorch/_aot_autograd/functional_utils.py

```diff
@@ -408,14 +408,13 @@ def assert_functional_graph(fx_g: torch.fx.Graph) -> int:
                 torch.ops.aten.set_.source_Tensor,
             ]:
                 suffix = True
-                # Can only copy_/set_ into an input, and can only do so once
```

Comment:
> can only do so once

@bdhirsh I believe even before the change in https://github.com/pytorch/pytorch/pull/130577, we are already secretly doing `.set_` into the same primal if the eager program is doing `.set_` multiple times into the same primal (it just didn't show up in the graph as so, because the 2nd set will look like `set__2 = .set_(set__1, Y)` instead of `set__2 = .set_(primal_X, Y)`)

After the change in https://github.com/pytorch/pytorch/pull/130577, doing `.set_` multiple times into the same primal will directly fail, due to the `placeholders.remove(n.args[0])` line below.

I feel that it might actually be okay to do multiple `.set_` into the same primal, but curious if XLA CI agrees..